app/domain/authentication/authn_k8s/kubectl_exec.rb

```diff
@@ -181,27 +199,44 @@ def copy(k8s_object_lookup:,
         container: 'authenticator')
         execute(
           k8s_object_lookup: k8s_object_lookup,
-          pod_namespace: pod_namespace,
-          pod_name: pod_name,
-          container: container,
-          cmds: ['tar', 'xvf', '-', '-C', '/'],
-          body: tar_file_as_string(path, content, mode),
-          stdin: true
+          pod_namespace:     pod_namespace,
+          pod_name:          pod_name,
+          container:         container,
+          cmds:              %w(sh),
+          body:              copy_script(path, content, mode),
+          stdin:             true
         )
       end
 
       private
 
-      def tar_file_as_string(path, content, mode)
-        tarfile = StringIO.new("")
-
-        Gem::Package::TarWriter.new(tarfile) do |tar|
-          tar.add_file(path, mode) do |tf|
-            tf.write(content)
-          end
-        end
-
-        tarfile.string
+      def copy_script(path, content, mode)
```

Comment:
This function is not really representative in name about what it does since we don't use `tar` anymore. This now "sets content of file at `path` to `content`" so it needs a better one.